File: 66. Plus One/Python/Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry = 1
        index = len(digits) - 1
        while carry and index >= 0:
            if digits[index] == 9:
                digits[index] = 0
            else:
                digits[index] += 1
                carry = 0
            index -= 1
        if digits[0] == 0:
            digits = [0] * (len(digits) + 1)
            digits[0] = 1
        return digits

File: 066. Plus One/C++/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        int carry = 1;
        int index = digits.size() - 1;
        while (carry == 1 && index >= 0) {
            if (digits[index] == 9) {
                digits[index] = 0;
            } else {
                digits[index]++;
                carry = 0;
            }
            index--;
        }
        if (digits[0] == 0) {
            digits.push_back(0);
            digits[0] = 1;
        }
        return digits;
    }
};